daphne_worker/src/durable/aggregate_store.rs

```diff
@@ -52,9 +61,21 @@ impl DurableObject for AggregateStore {
             }
 
             (DURABLE_AGGREGATE_STORE_GET, Method::Post) => {
-                let agg_share: DapAggregateShare =
-                    state_get_or_default(&self.state, "agg_share").await?;
-                Response::from_json(&agg_share)
+                let collected: bool = state_get_or_default(&self.state, "collected").await?;
+                if !collected {
+                    let agg_share: DapAggregateShare =
+                        state_get_or_default(&self.state, "agg_share").await?;
```

Comment:
It's posible to get `collected` and `agg_share` concurrently, which would save on latency:
```rust
let agg_share_future = state_get_or_default(&self.state, "agg_share");
let collected: bool = state_get_or_default(&self.state, "collected").await?;
if !collected {
    let agg_sahre: DapAggregateShare = agg_share_future.await?;
}
```

Reply:
Thanks for the suggestion!